source/Server/Ldap/LdapCredentialValidator.cs

```diff
@@ -50,35 +51,50 @@ public IResultFromExtension<IUser> ValidateCredentials(string username, string p
                 return ResultFromExtension<IUser>.ExtensionDisabled();
             }
 
-            if (username == null) throw new ArgumentNullException(nameof(username));
+            if (string.IsNullOrWhiteSpace(username))
+                return ResultFromExtension<IUser>.Failed("No username provided");
+
+            try
+            {
+                log.Verbose($"Validating credentials provided for '{username}'...");
 
-            log.Verbose($"Validating credentials provided for '{username}'...");
+                var validatedUser = ldapService.ValidateCredentials(username, password, cancellationToken);
 
-            var validatedUser = ldapService.ValidateCredentials(username, password, cancellationToken);
-            if (!string.IsNullOrWhiteSpace(validatedUser.ValidationMessage))
+                return string.IsNullOrWhiteSpace(validatedUser.ValidationMessage)
+                    ? GetOrCreateUser(validatedUser, cancellationToken)
+                    : ResultFromExtension<IUser>.Failed(validatedUser.ValidationMessage);
+            }
+            catch (Exception ex)
```

Comment:
Note this will block the informative error messages from my other PR, but I'll merge this into that one, and add a custom exception type to handle it specifically.